### PR TITLE
`sage.misc.package`: Remove deprecated code

### PR DESCRIPTION
**Pull request by mkoeppe**
Remove code deprecated in:
- #31013 (2022)
- #30747 (2020)
- #30607 (2020)

### :memo: Checklist

- [x] The title is concise and informative.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation and checked the documentation preview.

### CODE REVIEW

File: src/sage/misc/all.py [hunk 1]
```diff
@@ -40,8 +40,7 @@
 lazy_import('sage.misc.dist', 'install_scripts', deprecation=34259)
 lazy_import('sage.misc.trace', 'trace', deprecation=34259)
 lazy_import('sage.misc.package', ('installed_packages', 'is_package_installed',
-                                  'standard_packages', 'optional_packages',
-                                  'experimental_packages', 'package_versions'),
+                                  'package_versions'),
```
Comment (kiwifb) [inline src/sage/misc/all.py:43]: I suggest we remove `lazy_import('sage.misc.package',...)` altogether. Those functions are always explicitly imported and only in limited context (setup.py and feature.pkg_systems.py). `package_versions` is never ever called anywhere anymore which suggests it should be at least deprecated if not removed.

Comment (mkoeppe) [reply src/sage/misc/all.py:43]: The import is already deprecated, but it's short of 1 year of deprecation

Comment (kiwifb) [reply src/sage/misc/all.py:43]: Fair enough.

**Review by kiwifb — APPROVED** [review]
Getting rid of that stuff is good, wish we could get rid of more faster.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**